### PR TITLE
CDAP-12381 Make UnauthorizedException a runtime exception so that it gets propagated correctly

### PR DESCRIPTION
**Pull request by poornachandra**
JIRA - https://issues.cask.co/browse/CDAP-12381
Build - https://builds.cask.co/browse/CDAP-RUT1273-1

### CODE REVIEW

File: cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java [hunk 1]
```diff
@@ -1478,6 +1477,7 @@ private void deployDummyAppWithImpersonation(NamespaceMeta nsMeta, @Nullable Str
   public void afterTest() throws Exception {
     Authorizer authorizer = getAuthorizer();
 
+    SecurityRequestContext.setUserId(ALICE.getName());
```
Comment (rohitsinha54) [inline cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java:1480]: why this change

**Review by rohitsinha54 — COMMENTED** [review]
One question other than that LGTM

File: cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java (hunk 1 above)
Comment (poornachandra) [reply cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java:1480]: So that test failures get isolated. Otherwise a single test failing will cascade onto other tests during `afterTest()` since the principal may not be reset to ALICE who has privileges to clean up entities.

Comment (rohitsinha54) [reply cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java:1480]: 👍

**Comment by rohitsinha54** [comment]
LGTM

**Comment by poornachandra** [comment]
Build is green, merging.
